File: app/src/main/java/com/cabe/app/novel/model/NovelContent.java
package com.cabe.app.novel.model;

import android.text.TextUtils;

/**
 * 作者：沈建芳 on 2017/10/9 16:23
 */
public class NovelContent extends BaseObject {
    public String title;
    public String content;
    public String url;
    public String preUrl;
    public String nextUrl;
    public boolean flagLast;

    @Override
    public boolean equals(Object obj) {
        if(obj instanceof NovelContent) {
            NovelContent content = (NovelContent) obj;
            if(!TextUtils.isEmpty(title) && !TextUtils.isEmpty(content.title)
                    && !TextUtils.isEmpty(url) && !TextUtils.isEmpty(content.url)
                    && title.equals(content.title) && url.equals(content.url)) {
                return true;
            }
        }
        return false;
    }

    public String toString() {
        return title + "#" + url;
    }
}

File: app/src/main/java/com/cabe/app/novel/model/NovelContent.kt
package com.cabe.app.novel.model

import android.text.TextUtils

/**
 * 作者：沈建芳 on 2017/10/9 16:23
 */
class NovelContent : BaseObject() {
    @JvmField
    var title: String? = null
    @JvmField
    var content: String? = null
    @JvmField
    var url: String? = null
    @JvmField
    var preUrl: String? = null
    @JvmField
    var nextUrl: String? = null
    @JvmField
    var flagLast = false
    @JvmField
    var source = SourceType.X23US
    override fun equals(other: Any?): Boolean {
        if (other is NovelContent) {
            if (!TextUtils.isEmpty(title) && !TextUtils.isEmpty(other.title)
                    && !TextUtils.isEmpty(url) && !TextUtils.isEmpty(other.url)
                    && title == other.title && url == other.url) {
                return true
            }
        }
        return false
    }

    override fun toString(): String {
        return "$title#$url"
    }

    override fun hashCode(): Int {
        var result = title?.hashCode() ?: 0
        result = 31 * result + (content?.hashCode() ?: 0)
        result = 31 * result + (url?.hashCode() ?: 0)
        result = 31 * result + (preUrl?.hashCode() ?: 0)
        result = 31 * result + (nextUrl?.hashCode() ?: 0)
        result = 31 * result + flagLast.hashCode()
        result = 31 * result + source.hashCode()
        return result
    }
}